articles/cli.md

```diff
@@ -0,0 +1,169 @@
+## Bonsai Command-Line Interface (CLI)
+
+The Bonsai CLI (Command-Line Interface) makes it possible to launch workflows from the OS command-line tool. In addition to opening workflows with the default graphic interface, the CLI enables the use of other advanced features, such as "no-editor" mode, initialization of variable's values, and specifying layouts.
+
+
+&nbsp;
+___
+### Installation and basic use
+Bonsai CLI can be called by targeting the ```Bonsai.exe``` executable. To make sure you are able to call Bonsai from the command line:
+
+```cmd
+> pathToBonsaiExe\Bonsai.exe
+```
+Alternatively, by adding ```pathToBonsaiExe``` to your System Path, simply:
+
+```cmd
+> Bonsai
+```
+
+This command will open the default Bonsai editor.
+
+___
+&nbsp;
+### Opening a Workflow
+
+To open an existing workflow (i.e., ```myWorkflow.bonsai```), use the syntax:
+
+##### Example:
+
+```cmd
+> Bonsai myWorkflow.bonsai
+> Bonsai "C:\Users\User\Desktop\Example\myWorkflow.bonsai"
+```
+___
+&nbsp;
+### Starting a Workflow
+
+Adding additional flags will change the output behavior of the Bonsai CLI. For instance, to launch and immediately start a workflow from the command line we add the ```--start``` flag to the previous command:
+
+
+##### Example:
+```cmd
+> Bonsai myWorkflow.bonsai --start
+```
+
+___
+&nbsp;
+### Starting a Workflow without the editor
+
+
+For some applications, such as batch processing, the user might not be interested in having access to the UI. This can be achieved using the ```--no-editor``` flag. Note that the ```--start``` flag behavior is implicit when running with ```--no-editor``` mode.
+
+
+##### Example:
+```cmd
+> Bonsai myWorkflow.bonsai --no-editor
+```
+
+Upon successful launch of your workflow, a Bonsai icon will appear on your system tray that can be used to stop the workflow or open other visualizers.
+
+(![SystemTray Example](~/images/Article_CLI_SystemTrayBonsai.png)
+
+It is worth noting that only assigned visualizers will be available on this list. To assign a visualizer:
+- Open the workflow in editor mode;
+- Right click on top of the node of interest;
+- "Show Visualizer" ➡ Select a visualizer other than "None"
+
+(![Select visualizer](~/images/Article_CLI_Select_visualizer.png)
+
+
+___
+&nbsp;
+### Passing parameter values to workflows
+
+One of the most useful features of the CLI is the ability to, using the same workflow, passing different values to different nodes before running it. Only "externalized properties" can be set from the command line.
+
+##### To externalize a property:
+- Open the workflow in editor mode;
+- Right click on top of the node of interest;
+- "Externalize Property" ➡ Select the property of interest
+
+(![Externalize Property Example 1](~/images/Article_CLI_extern_prop_example.png)
+
+- A new node will appear with a dashed line connected to the node of interest.
+- No two externalized properties, in the same level, are allowed to share the same name.
```

Comment:
What is meant by "in the same level"?

Reply:
I was debating what word to use here. Basically, once you group nodes, you need to use the dot syntax, but it also allows you to have properties with the same name as long as they are not inside the same group. I dont want to use the word nested, since that is somewhat related with ```NestedWorkflow```. I could simply write:

"No two externalized properties, in the same group, are allowed to share the same name."

Would this be clearer?

Reply:
An alternative would be to simply link directly to the article explaining how externalized properties work in detail, and just describe here how the CLI interacts with these properties.

Reply:
Should I just leave it as is for now (check the latest commit) and perhaps update it later once such an article has been written?